docs/v2_proposal.rst

```diff
@@ -0,0 +1,39 @@
+=======================
+OCX-Library V2 Proposal
+=======================
+
+The intention of this library is to act as a wrapper around the underlying client libraries.
+A user will import all of the "python-[service]client" libraries they would like to use, in addition to ours. 
+Our library relies on the existance of the other client libraries in the python environment.
+
+Code control flow:
+------------------
+* First call
+    - initialize x keystone clients
+    - initialize x service clients, one for each keystone client
+    - call the function with kwargs on the x service clients
+* Every subsequent call, until clients time out
+    - retrieve the service clients stored in the user's running software 
+    - call the function with kwargs on the x service clients
+* Once clients time out, we inform the user, and ask if they would like to re-initialize the clients
+
+Necessary Functions:
+--------------------
+* init_keystone_clients(auth_url_list, user_name, password)
+    - return keystone_client_list
+* init_service_clients(keystone_client_list, service)
+    - return service_client_list [list of nova clients, neutron clients, etc...]
+* call_service(service_client_list, function, args, kwargs)
+    - return aggregate_of_return_values
+
+Issues to be addressed:
+-----------------------
+* Currently every call is being done on every endpoint you want to initialize... should we try to limit this scope? At which point? 
+    - Some easy way to filter the list of their clients passed to a service function call?
```

Comment:
As for this, since we want to use this wrapper library for OCX, we would have fixed endpoints, for example, harvard cluster is "http://140.247.152.207:5000/v2.0" and we would also have similar ones for different other clusters. 

In all, limiting entry of endpoints from users would be something that necessary for us to implement